src/app/location/location.component.html

```diff
@@ -0,0 +1,14 @@
+<div class="container">
+  <h1>Location Test</h1>
+  <form #cityForm="ngForm" (ngSubmit)="onSubmit(cityForm)">
+    <div class="form-group">
+      <label for="city">City: </label>
+      <input type="text" id="city" name="city" required ngModel />
+      <div [hidden]="city" class="alert">City is required</div>
+    </div>
+    <button type="submit" class="btn">City location</button>
+    <button type="button" class="btn" (click)="getUserLocation()">
+      GPS location
+    </button>
```

Comment:
please use [ng components](https://www.primefaces.org/primeng/showcase/#/inputtext) 

Not sure if this a part of this task ;p